resources/mediaplayer/src/main/java/org/mobicents/media/server/impl/resource/mediaplayer/audio/AudioCacheECache.java

```diff
@@ -0,0 +1,42 @@
+package org.mobicents.media.server.impl.resource.mediaplayer.audio;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+
+import org.apache.commons.io.IOUtils;
+import org.ehcache.Cache;
+import org.ehcache.CacheManager;
+import org.ehcache.config.builders.CacheConfigurationBuilder;
+import org.ehcache.config.builders.CacheManagerBuilder;
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.config.units.MemoryUnit;
+
+/**
+ * Created by achikin on 5/9/16.
+ */
+public class AudioCacheECache implements AudioCache {
+    private CacheManager cacheManager;
+
+    public AudioCacheECache(int size) {
+        cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
+                .withCache("preConfigured",
+                        CacheConfigurationBuilder.newCacheConfigurationBuilder(String.class, byte[].class,
+                                ResourcePoolsBuilder.newResourcePoolsBuilder().heap(size, MemoryUnit.MB))
+                                .build())
+                .build(true);
+    }
+
+    public Cache getCache() {
+        return cacheManager.getCache("preConfigured", String.class, byte[].class);
+    }
+
+    public InputStream getStream(URL uri) throws IOException {
+        Cache<String, byte[]> cache = getCache();
+        if (!cache.containsKey(uri.toString())) {
+            cache.put(uri.toString(), IOUtils.toByteArray(uri.openStream()));
```

Comment:
I know about arguments evaluation.

I was wondering if the Cache API allowed some kind of entry interface as http://www.ehcache.org/apidocs/3.0.2/org/ehcache/Cache.Entry.html . Apparently this is not longer available in version 3, so no worries (it was before http://www.ehcache.org/apidocs/2.9/index.html).

Its not big issue anyway, just checking  if putIfAbsent was possible, since it uses to deliver better behaviour in terms of concurrency. So, what would be the behaviour if two threads happen to put the same URI in the map?. I checked the cache allows concurrency, but, is the cache eviction policy evaluated twice in that case...?